### PR TITLE
fix: Correctly link errors in problems tab to problem location in file

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
Fixes an issue where the URI for the file was built incorrectly, which prevented clicking on the problem in the problems tab from linking to the correct location in the file.

### What issues does this PR fix or reference?
#4194, #4228 @[W-11350093]@

### Functionality Before

https://user-images.githubusercontent.com/46458081/176308232-ac30c395-36e0-40f6-991d-99c040869dcc.mp4

### Functionality After

https://user-images.githubusercontent.com/46458081/176308217-3750e4a8-0447-4811-a143-f70f6cf2c308.mp4

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts [hunk 1]
```diff
@@ -13,6 +13,18 @@ import {
...
+export function getFileUri(
+  workspacePath: string,
+  filePath: string,
+  defaultErrorPath: string
+): string {
+  const resolvedFilePath = filePath.includes(workspacePath)
+    ? filePath
+    : path.join(workspacePath, filePath);
+  // source:deploy sometimes returns N/A as filePath
+  return filePath === 'N/A' ? defaultErrorPath : resolvedFilePath;
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts:25]: Should we track when this happens?

File: packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts [hunk 2]
```diff
@@ -71,6 +75,7 @@ describe('Diagnostics', () => {
     const testDiagnostics = languages.getDiagnostics(
       Uri.file(path.join(workspacePath, resultItem.filePath))
     );
+    console.log(testDiagnostics);
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts:78]: nit: probably should delete this console.

File: packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts (hunk 1 above)
Comment (gbockus-sf) [reply packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts:25]: nit: make 'N/A' a constant

File: packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts [hunk 3]
```diff
@@ -227,4 +232,12 @@ describe('Diagnostics', () => {
...
     expect(testDiagnostics[1].range).to.deep.equal(testRange1);
   });
+
+  it('Should not duplicate the workspace path when constructing the fileUri', () => {
+    const absoluteFilePath = `${workspacePath}/src/classes/Testing.cls`;
+    const fileUri = getFileUri(workspacePath, absoluteFilePath, '');
+    const regEx = new RegExp(workspacePath, 'g');
+    const count = (fileUri.match(regEx) || []).length;
+    expect(count).to.equal(1);
+  });
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/test/vscode-integration/diagnostics/index.test.ts:242]: A test of the N/A case would be good.

**Review by gbockus-sf — COMMENTED** [review]
good find.  A couple of nitty suggestions.

File: packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts (hunk 1 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts]: Not sure if we should track this.  I posed a question internally in Slack to see if this "N/A" response is still relevant...

**Review by gbockus-sf — APPROVED** [review]
Sweet.  Good first day back fix :)

**Pull request merged**